p2p/src/network/noise/p2p_network_noise_actions.rs

```diff
@@ -75,7 +75,16 @@ impl From<P2pNetworkNoiseAction> for crate::P2pAction {
 }
 
 impl redux::EnablingCondition<P2pState> for P2pNetworkNoiseAction {
-    fn is_enabled(&self, _state: &P2pState, _time: redux::Timestamp) -> bool {
+    fn is_enabled(&self, state: &P2pState, _time: redux::Timestamp) -> bool {
+        let Some(_noise_state) = state
+            .network
+            .scheduler
+            .connection_state(self.addr())
+            .and_then(|state| state.noise_state())
```

Comment:
maybe `.is_none()` instead of else?

Reply:
I left it as `_state` just so it is easier to add state later but I can change it to `.is_none`